### PR TITLE
Blog post about Kafka commit strategies

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: _posts/2020-11-03-kafka-commit-strategies.adoc [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+author: cescoffier
+---
+
+In a https://quarkus.io/blog/kafka-failure-strategy/[previous blog post], we have looked at failure strategies provided by the Reactive Messaging Kafka connector. 
+But, imagine it's our lucky day, and for once it worked.
+We should inform Kafka that the processing succeeded. 
+In Kafka terminology, we call this: _offset commit_.
+This post covers the different strategies to commit offsets with the Reactive Messaging Kafka connector.
+
+## Kafka consumer group and Offsets
```
Comment (kenfinnigan) [inline _posts/2020-11-03-kafka-commit-strategies.adoc:16]: Any reason for "consumer group" not being capitalized but "Offsets" is?

File: _posts/2020-11-03-kafka-commit-strategies.adoc [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+So, with Kafka, you can identify an individual record using a `<topic, partition, offset>` tuple. 
+
+image:/assets/images/posts/kafka-commit-strategies/topics-partitions.png[]
+
+When an application consumes messages from Kafka, it uses a Kafka consumer.
+With this consumer, it polls batches of messages from a specific topic, for example, `movies` or `actors`.
+Retrieved messages belong to partitions assigned to this consumer. 
+And that aspect is essential. 
+
+Consumers belong to a consumer group, identified with a name (`A` and `B` in the picutre above).
```
Comment (kenfinnigan) [inline _posts/2020-11-03-kafka-commit-strategies.adoc:36]: s/picutre/picture

File: _posts/2020-11-03-kafka-commit-strategies.adoc [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+So, it gets the records from all three partitions.
+Consequently (ignore rebalance or other subtilities for now), each record from a topic is only received once per consumer group, by a specific consumer from that group. 
+
+To orchestrate each consumer group's progress, each consumer periodically informs the broker of its current position - the last processed offset. 
+It commits the offset, indicating that all the previous records from that partition have been processed. 
+So, if a consumer stops and comes back later, it restarts from the last committed position (if assigned to that partition again). 
+Note that this behavior is configurable. 
+
+What's important to notice is the periodic aspect of the commit. 
+Offset commit is expensive, and to enhance performances, we should not commit the offset after each processed record.
```
Comment (kenfinnigan) [inline _posts/2020-11-03-kafka-commit-strategies.adoc:56]: maybe drop the "s" on performances?

File: _posts/2020-11-03-kafka-commit-strategies.adoc [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+## What if the message's processing is asynchronous
+
+If the message processing is asynchronous (offloaded to another thread, use non-blocking I/O...), failures may not interrupt the while loop from above. 
+Failure happens asynchronously, outside the polling thread.
+If the `poll` method gets called again despite a failed processing, and _auto-commit_ is still enabled, we may commit offsets while something wrong happened. 
+If some processing of previously retrieved records is not completed yet, while this auto commit happens, it may consider the record as processed correctly, but the outcome is unknown at that point. 
+
+So to handle these case, we can disable the auto-commit and switch to manual commit. 
+In this case, it's the application's responsibility to commit the offsets regularly. 
+So, the application needs to track the polled records, their processing, failures, and periodically commits.
```
Comment (kenfinnigan) [inline _posts/2020-11-03-kafka-commit-strategies.adoc:96]: Should "commits" be plural or singular?

File: _posts/2020-11-03-kafka-commit-strategies.adoc [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+It might not look too tricky, but actually, it can become quite challenging.
+Again, in asynchronous scenarios,  you may complete the processing of messages in various orders.
```
Comment (kenfinnigan) [inline _posts/2020-11-03-kafka-commit-strategies.adoc:98]: Extra space after the second comma

File: _posts/2020-11-03-kafka-commit-strategies.adoc [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+You can enable this strategy by configured enabled-auto-commit to true:
+
+[source, properties]
+----
+mp.messaging.incoming.my-channel.connector=smallrye-kafka
+mp.messaging.incoming.my-channel.enable.auto.commit=true
+----
+
+IMPORTANT: Be aware that starting Quarkus 1.9, auto commit is disabled by default. So you need to explicitly enables it.
```
Comment (kenfinnigan) [inline _posts/2020-11-03-kafka-commit-strategies.adoc:166]: s/enables/enable

**Review by kenfinnigan — CHANGES_REQUESTED** [review]

File: _posts/2020-11-03-kafka-commit-strategies.adoc (hunk 4 above)
Comment (cescoffier) [reply _posts/2020-11-03-kafka-commit-strategies.adoc:96]: It's a verb in this case. I added `commits the offsets`

File: _posts/2020-11-03-kafka-commit-strategies.adoc (hunk 3 above)
Comment (cescoffier) [reply _posts/2020-11-03-kafka-commit-strategies.adoc:56]: In French, we can count them :-D

**Review by kenfinnigan — APPROVED** [review]

**Comment by cescoffier** [comment]
Thanks @kenfinnigan !

**Pull request merged**